src/nccl_ofi_rdma.c

```diff
@@ -1649,53 +1715,12 @@ static int ofi_process_cq(nccl_net_ofi_rdma_ep_t *ep)
 				if (OFI_UNLIKELY(ret != 0))
 					goto exit;
 			} else if (OFI_UNLIKELY(rc == -FI_EAVAIL)) {
-				rc = fi_cq_readerr(rail->cq, &err_buffer, 0);
-				if (OFI_UNLIKELY(rc == -FI_EAGAIN)) {
-					/*
-					 * Error not available yet.
-					 * fi_cq_read will keep returning -FI_EAVAIL so just bail out and try again later.
-					 */
+				ret = process_err_completion(ep, rail_id);
+				if (ret == 0)
+					/* Error entry not available yet */
```

Comment:
Reading `process_err_completion`, I'm not actually sure how this case can happen.

Reply:
If for some reason fi_cq_readerr() comes back with EAGAIN, we'll get here. I too think the chances of this happening is slim (as the provider should have queued up the error entry before actually blocking the primary CQ).. but I checked the man pages and EAGAIN is a valid ret code for _readerr() still so definition doesn't force a provider to queue up the err entry first.

Reply:
Makes sense, I missed the return 0 on `EAGAIN`. Thanks.